### PR TITLE
Enums

### PR DESCRIPTION
**Pull request by domoritz**
https://basarat.gitbooks.io/typescript/content/docs/enums.html

### CODE REVIEW

File: src/Encoding.ts [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 import * as vlEncDef from './encdef';
 import * as vlEnc from './enc';
 import * as schema from './schema/schema';
 import * as schemaUtil from './schema/schemautil';
 
-export class Encoding {
+export default class Encoding {
   _data: any;
-  _marktype: any;
+  _marktype: string;  // TODO: replace with enum
```
Comment (kanitw) [inline src/Encoding.ts]: This should refer to MarkType interface for the schema

File: test/Encoding.test.js [hunk 2]
```diff
@@ -1,15 +1,11 @@
-'use strict';
-
-var expect = require('chai').expect;
-
-var Encoding = require('../src/Encoding').Encoding;
-
+var chai_1 = require('chai');
```
Comment (kanitw) [inline test/Encoding.test.js]: Why this weird `chai_1` pattern?

Comment (domoritz) [inline test/Encoding.test.js]: Compiled from ts

Comment (kanitw) [inline test/Encoding.test.js]: So I guess we check in both ts, js because we haven't migrated test to run on TS?

Comment (domoritz) [inline test/Encoding.test.js]: No, just this one because I forgot to delete it. Sorry.

**Comment by domoritz** [comment]
@kanitw ready to merge

**Pull request merged**